plugins/mana/plugin.go

```diff
@@ -678,6 +685,29 @@ func pruneConsensusEventLogsStorage() {
 	consensusEventsLogsStorageSize.Sub(uint32(len(entriesToDelete)))
 }
 
+func cleanupManaVectors() {
+	toRemove := make(map[mana.Type][]identity.ID)
+	types := []mana.Type{mana.AccessMana, mana.ConsensusMana}
+	for _, vecType := range types {
+		baseManaVectors[vecType].ForEach(func(id identity.ID, baseMana mana.BaseMana) bool {
+			switch vecType {
+			case mana.ConsensusMana:
+				if baseMana.EffectiveValue() < minEffectiveMana && baseMana.BaseValue() == 0 {
+					toRemove[vecType] = append(toRemove[vecType], id)
+				}
+			case mana.AccessMana:
+				if baseMana.EffectiveValue() < minEffectiveMana && baseMana.BaseValue() < minBaseMana1 {
+					toRemove[vecType] = append(toRemove[vecType], id)
+				}
+			}
+			return true
+		})
+	}
+	for _, vecType := range types {
+		baseManaVectors[vecType].Remove(toRemove[vecType])
+	}
```

Comment:
@acha-bill let's make the removal atomic, becasue now it can happen that between determining which nodes to remove and actually removing them a pledge happens

Reply:
done